external-crates/move/crates/move-compiler/src/typing/translate.rs

```diff
@@ -4277,16 +4287,33 @@ fn expand_macro(
                     sp(b.loc, TS::Bind(b, lvalue_ty, Box::new(e)))
                 })
                 .collect();
+            let by_value_num = seq.len();
             // add the body
             let body = exp(context, body);
             let ty = body.ty.clone();
             seq.push_back(sp(body.exp.loc, TS::Seq(body)));
             let use_funs = N::UseFuns::new(context.current_call_color());
-            let e_ = TE::Block((use_funs, seq));
+            let e_ = TE::Block(
+                (use_funs, seq),
+                /* inlined macro */
+                Some((
+                    mod_ident,
+                    f,
+                    m,
+                    type_args
+                        .into_iter()
+                        .map(|mut t| {
+                            expand::type_(context, &mut t);
```

Comment:
Without this we are still seeing `TVar`s here and IDE does not know what to do with them...

Reply:
We cannot do this because it will break inference.
You should not have `TVars` at this location if the macro type checked

Reply:
It is unsurprising that this breaks, as the variables may not be sufficiently grounded at this point.

Reply:
Moved this to the right place in `expand.rs`